File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = ''
    stk=[]
    for num in number:
        while k>0 and stk and stk[-1]<num:
            stk.pop()
            k-=1
        stk.append(num)
        
    if k>0:
        stk=stk[:-k]
        
    return ''.join(stk)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>
#include <stack>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    stack<char> stk;
    for(auto n:number){
        if(stk.empty() || stk.top()>=n || k<=0){
            stk.push(n);
        }
        else{
            while(!stk.empty() && stk.top()<n && k>0){
                stk.pop();
                k--;
            }
            stk.push(n);
        }
        
    }
    while(k>0){
        stk.pop();
        k--;
    }
    while(!stk.empty()){
        answer+=stk.top();
        stk.pop();
    }
    reverse(answer.begin(), answer.end());
    
    return answer;
}